### PR TITLE
Fixing parsing non-json response

### PR DESCRIPTION
**Pull request by patrykbuzowiczsabre**

### CODE REVIEW

File: src/actions/api.ts [hunk 1]
```diff
@@ -51,18 +51,18 @@ export interface FetchActions<Success, Error> {
 
 type DeferredResult = (dispatch: Dispatch, getState: () => ReduxState) => Action | DeferredResult;
 
-const transformErrorToReject = (result: Response | Response[]) => new Promise((resolve, reject) => {
-  if (Array.isArray(result)) {
-    const jsonPromises = result.map(r => r.json());
-    const data = Promise.all(jsonPromises);
+const transformErrorToReject = (response: Response | Response[]) => new Promise((resolve, reject) => {
+  if (Array.isArray(response)) {
+    const resultPromises = response.map(getResult);
```
Comment (wojciech-kumon) [inline src/actions/api.ts]: Why don't we check `response.ok` first? https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch#Checking_that_the_fetch_was_successful

Comment (patrykbuzowiczsabre) [reply src/actions/api.ts]: we do check it below within `some`, line 59

Comment (patrykbuzowiczsabre) [reply src/actions/api.ts]: at this point we want the response payload regardless of whether successful or not

Comment (wojciech-kumon) [reply src/actions/api.ts]: ok, I forgot that we use response body on failure

**Pull request merged**